src/modules/message/components/signedMessage/index.js

```diff
@@ -1,40 +1,42 @@
 import React, { useEffect, useRef, useState } from 'react';
 import CopyToClipboard from 'react-copy-to-clipboard';
+import { withRouter } from 'react-router';
 import { useTranslation } from 'react-i18next';
 import BoxContent from 'src/theme/box/content';
 import BoxFooter from 'src/theme/box/footer';
 import Illustration from 'src/modules/common/components/illustration';
 import { getDeviceType } from '@wallet/utils/hwManager';
 import { AutoResizeTextarea } from 'src/theme';
 import { SecondaryButton, PrimaryButton } from 'src/theme/buttons';
+import { removeSearchParamsFromUrl } from 'src/utils/searchParams';
 import styles from './signedMessage.css';
 
 const Error = ({ t, hwInfo }) => {
   const deviceType = getDeviceType(hwInfo?.deviceModel);
 
   return (
-    <BoxContent className={`${styles.noPadding} ${styles.statusWrapper}`}>
+    <BoxContent className={styles.statusWrapper}>
       <Illustration name={`${deviceType}HwRejection`} />
       <h5>{t('Transaction aborted on device')}</h5>
       <p>{t('You have cancelled the transaction on your hardware wallet.')}</p>
     </BoxContent>
   );
 };
 
-const Success = ({ t, signature, copied, copy, prevStep, onPrev }) => (
+const Success = ({ t, signature, copied, copy, history, onPrev }) => (
   <>
-    <BoxContent className={styles.noPadding}>
+    <BoxContent>
       <AutoResizeTextarea className={`${styles.result} result`} value={signature} readOnly />
     </BoxContent>
     <BoxFooter direction="horizontal">
       <SecondaryButton
         onClick={() => {
+          removeSearchParamsFromUrl(history, ['modal'], true);
           onPrev?.();
```

Comment:
Line 35 should come before line 34 to prevent trying to modify a state after a component has been unmounted